AutocompleteClientTests/FW/Logic/Worker/ConstructorIOBrowseFacetOptionsTests.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  ConstructorIOBrowseFacetOptionsTests.swift
+//  AutocompleteClientTests
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import ConstructorAutocomplete
+import XCTest
+
+class ConstructorIOBrowseFacetOptionsTests: XCTestCase {
+
+    var constructor: ConstructorIO!
+
+    override func setUp() {
+        super.setUp()
+        self.constructor = ConstructorIO(config: TestConstants.testConfig)
+    }
+
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+
+    func testBrowseFacetOptions_CreatesValidRequest() {
+        let query = CIOBrowseFacetOptionsQuery(facetName: "price")
+
+        let builder = CIOBuilder(expectation: "Calling Browse Facet Options should send a valid request.", builder: http(200))
+        stub(regex("https://ac.cnstrc.com/browse/facet_options?c=\(kRegexVersion)&facet_name=price&i=\(kRegexClientID)&key=key_OucJxxrfiTVUQx0C&s=\(kRegexSession)"), builder.create())
+
+        self.constructor.browseFacetOptions(forQuery: query, completionHandler: { response in })
+        self.wait(for: builder.expectation)
+    }
+
+    func testBrowseFacetOptions_WithValidRequest_ReturnsNonNilResponse() {
+        let expectation = self.expectation(description: "Calling Browse Facet Options with valid parameters should return a non-nil response.")
+
+        let query = CIOBrowseFacetOptionsQuery(facetName: "price")
+
+        let dataToReturn = TestResource.load(name: TestResource.Response.browseFacetOptionsJSONFilename)
+        stub(regex("https://ac.cnstrc.com/browse/facet_options?c=\(kRegexVersion)&facet_name=price&i=\(kRegexClientID)&key=key_OucJxxrfiTVUQx0C&s=\(kRegexSession)"), http(200, data: dataToReturn))
+
+        self.constructor.browseFacetOptions(forQuery: query, completionHandler: { response in
+            XCTAssertNotNil(response.data, "Calling Browse Facet Options with valid parameters should return a non-nil response.")
+
+            XCTAssertNotNil(response.data?.facets)
+            XCTAssertNotNil(response.data?.facets[0].options)
+            XCTAssertNotNil(response.data?.resultID)
```

Comment:
Could we do some assertions against actual values to make sure the parser is working as expected? Since the response is static, it would make sense to check them here instead of the integration tests (where we just check for non-nil values).